client/interface.go

```diff
@@ -10,6 +10,8 @@ import (
 
 // Client represents the drand Client interface.
 type Client interface {
+	// String returns the name of this client instance.
+	String() string
```

Comment:
Note that this overloads the built-in [`Stringer`](https://tour.golang.org/methods/17) interface already used by `fmt` and others. Intentional?

Reply:
I think the implementations can accurately serve both purposes. The clients didn't implement Stringer prior to this PR anyway.

Reply:
Makes sense.

I agree with your choice of adding String() methods and logging failures at the optimizing layer.
I think my preference would not add `String()` in the `Client` interface here, since this is our contract with downstream consumers, and `String` isn't key to that interface. The optimizer can `fmt.Sprintf()` clients, and String will get invoked when present even if it isn't part of this interface.

Reply:
Sounds good. I'll make this change.